### PR TITLE
fix: :bug: add error handling for docker service startup failures

### LINKED ISSUE #1877: `pnpm local` should show errors
### Preliminary Checks

- [X] I have reviewed https://unkey.com/docs for existing features that would solve my problem

- [X] I have searched for existing feature requests: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a feature request directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Is your feature request related to a problem? Please describe.

Currently errors are swallowed and give no indication of what went wrong

### Describe the solution

It should print the error

### Describe alternatives you have considered (if any)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DeepaPrasanna**
## What does this PR do?

Fixes #1877 

Updated the code to handle errors when starting docker services. If the service startup fails, an error is thrown with a descriptive message indicating which service encountered the issue.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Run `pnpm local` without running docker.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [ ] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [ ] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

**Comment by chronark** [comment]
This is better than what we currently have, but it would be nice if we can return the original error, so the user actually has something to work with

**Comment by DeepaPrasanna** [comment]
The error was only a code with value 1, so that's why I thought it will be unclear to return that value.

**Comment by chronark** [comment]
docker definitely returns more descriptive errors than that

**Comment by DeepaPrasanna** [comment]
Hmm, maybe I missed. I will have a look at it again.

**Comment by DeepaPrasanna** [comment]
Is this okay? I am now throwing the original error

Untitled

**Comment by chronark** [comment]
that doesn't help a developer. we need to return the actual error from the command that ran, not the error from the `node:child_process`

**Comment by DeepaPrasanna** [comment]
did u meant the `stderr` from the command itself?

**Comment by DeepaPrasanna** [comment]
In that case, It was empty. I will attach a screenshot. 
image

That why I went with `error` even though that doesnt helps much

**Comment by DeepaPrasanna** [comment]
shall i return `stdout` along with `error` ?

**Comment by chronark** [comment]
yeah that makes sense I think

**Comment by DeepaPrasanna** [comment]
Is this okay?

Screenshot 2024-07-18 223255

**Comment by chronark** [comment]
👍

**Comment by DeepaPrasanna** [comment]
I pushed the required changes.

File: tools/local/src/docker.ts [hunk 1]
```diff
@@ -16,7 +16,13 @@ export async function startContainers(services: Array<string>) {
 
 async function run(cmd: string, opts: { cwd: string }) {
   await new Promise((resolve, reject) => {
-    const p = exec(cmd, opts);
+    const p = exec(cmd, opts, (error, stdout) => {
+      if (error) {
+        // biome-ignore lint/suspicious/noConsoleLog: logging seems the best option here
+        console.log(stdout);
```
Comment (chronark) [inline tools/local/src/docker.ts:22]: it's an error, why don't we use `console.error`?

Comment (DeepaPrasanna) [reply tools/local/src/docker.ts:22]: Yes, that should have been used

**Review by chronark — APPROVED** [review]

**Pull request merged**